galsim/utilities.py

```diff
@@ -1331,3 +1331,46 @@ def rand_with_replacement(n, n_choices, rng, weight=None, _n_rng_calls=False):
         return index, n_rng_calls
     else:
         return index
+
+
+def check_share_file(filename, subdir):
+    """Find SED or Bandpass file, possibly adding share dir or raising deprecation warning if old
+    share dir was specified.
+    """
+    from .deprecated import depr
+    import os
+
+    if os.path.isfile(filename):
+        return True, filename
+
+    new_filename = os.path.join(galsim.meta_data.share_dir, subdir, filename)
+    if os.path.isfile(new_filename):
+        return True, new_filename
+
+    dirname, basename = os.path.split(filename)
+    new_filename = os.path.join(dirname, subdir, basename)
+    if os.path.isfile(new_filename):
+        depr("Filename os.path.join(galsim.meta_data.share_dir, {0})".format(basename),
+             1.5,
+             "os.path.join(galsim.meta_data.share_dir, '{0}', {1})".format(subdir, basename))
+        return True, new_filename
+
+    return False, ''
+
+
+# http://stackoverflow.com/a/6849299
+class lazy_property(object):
+    """
+    meant to be used for lazy evaluation of an object attribute.
+    property should represent non-mutable data, as it replaces itself.
+    """
+    def __init__(self, fget):
+        self.fget = fget
+        self.func_name = fget.__name__
+
+    def __get__(self, obj, cls):
+        if obj is None:
+            return None
```

Comment:
Looking into this a bit, I think None is the wrong thing to return:
```
>>> galsim.Gaussian.sigma
<property object at 0x10578f940>
>>> print galsim.Gaussian.noise
None
```
In the Python docs, they don't return `None` for this branch. they return self, which IMO looks a bit more consistent:
```
>>> galsim.Gaussian.noise
<galsim.utilities.lazy_property object at 0x10f6d7490>
```
Without the branch entirely, it raises an error
```
>>> galsim.Gaussian.noise
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Mike/lib/python2.7/site-packages/galsim/utilities.py", line 1373, in __get__
    setattr(obj, self.func_name, value)
AttributeError: 'NoneType' object has no attribute 'noise'
```